src/dosext/mfs/mfs.c

```diff
@@ -957,8 +958,11 @@ mfs_redirector(void)
 }
 
 int
-mfs_inte6(void)
+mfs_inte6(int fromDOS)
 {
+  if (fromDOS && WORD(REGS.ebx) == DOS_SUBHELPER_MFS_REDIR_INIT)
+    emufs_loaded = TRUE;
```

Comment:
Oh, why is this?
Please remove this nasty fromDOS stuff.
How about just not removing the init
at `bx=0`? You can just put it there.
And it is actually much better because
this init can return the "already initialize"
error, so you don't mess with version
check.